src/components/Comments.jsx

```diff
@@ -1,36 +1,86 @@
 import { useState, useEffect } from "react";
 import { useParams } from "react-router-dom";
 import { getComments } from "../utils/api";
+import { postComment } from "../utils/api";
+import { deleteComment } from "../utils/api";
 
 export default function Comments() {
   const [comments, setComments] = useState([]);
+
   const { article_id } = useParams();
 
   useEffect(() => {
-    getComments(article_id).then((body) => {
-      setComments(body);
-    });
+    getComments(article_id).then((body) => setComments(body));
   }, []);
 
+  function handleOpenAddCommentDlg() {
+    const addCommentDlg = document.getElementById("addCommentDlg");
+    addCommentDlg.showModal();
+  }
+
+  function handleCloseAddCommentDlg() {
+    const addCommentDlg = document.getElementById("addCommentDlg");
+    comment.value = "";
+    addCommentDlg.close();
+  }
+
+  function handlePostComment() {
+    const newComment = { body: comment.value, username: "cooljmessy" };
+    if (newComment.body)
+      postComment(article_id, newComment).then((comment) => {
+        handleCloseAddCommentDlg();
+        setComments([comment, ...comments]);
+      });
+  }
+
+  function handleDeleteComment(deletedComment) {
+    deleteComment(deletedComment.comment_id).then(() => {
+      const updatedComments = comments.filter((comment) => comment.comment_id !== deletedComment.comment_id);
+      setComments(updatedComments);
+    });
```

Comment:
Nice. You may want to add a deleting in progress message and a delete error message too.